### PR TITLE
A few bug fixes for Repository resource.

### PR DESCRIPTION
**Pull request by apurvmahajan**
*Description of changes:*

This PR fixes 2 bugs with the Repository resource.
* Allow updates to resource if RepositoryPolicy and LifeCycle Policy resource is not present. Currently, the updates fail if the user does not have the appropriate permissions to delete/update the policy.
* Tag update fails for key in case of key collisions.

*Testing*
* mvn package
* CFN test

```
handler_create.py::contract_create_delete                                                                                                                   [  6%]
handler_create.py::contract_invalid_create PASSED                                                                                                           [ 12%]
handler_create.py::contract_create_duplicate PASSED                                                                                                         [ 18%]
handler_create.py::contract_create_read_success PASSED                                                                                                      [ 25%]
handler_create.py::contract_create_list_success PASSED                                                                                                      [ 31%]
handler_delete.py::contract_delete_read PASSED                                                                                                              [ 37%]
handler_delete.py::contract_delete_list PASSED                                                                                                              [ 43%]
handler_delete.py::contract_delete_update PASSED                                                                                                            [ 50%]
handler_delete.py::contract_delete_delete PASSED                                                                                                            [ 56%]
handler_delete.py::contract_delete_create PASSED                                                                                                            [ 62%]
handler_misc.py::contract_check_asserts_work PASSED                                                                                                         [ 68%]
handler_read.py::contract_read_without_create PASSED                                                                                                        [ 75%]
handler_update.py::contract_update_read_success PASSED                                                                                                      [ 81%]
handler_update.py::contract_update_list_success PASSED                                                                                                      [ 87%]
handler_update_invalid.py::contract_update_create_only_property PASSED                                                                                      [ 93%]
handler_update_invalid.py::contract_update_non_existent_resource PASSED                                                                                     [100%]

============================================================== 16 passed in 680.47s (0:11:20) ====================================================================
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-ecr-repository/src/test/java/software/amazon/ecr/repository/ConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.junit.jupiter.api.Test;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ConfigurationTest {
+
+    @Test
+    void testMergeDuplicateKeys() {
```
Comment (bradgr) [inline aws-ecr-repository/src/test/java/software/amazon/ecr/repository/ConfigurationTest.java:13]: Would like to start being more descriptive on test naming. We can also remove the 'test' part, I think, so something like '{condition}_{expectedBehavior}'

File: aws-ecr-repository/src/main/java/software/amazon/ecr/repository/Configuration.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public JSONObject resourceSchemaJSONObject() {
         if (resourceModel.getTags() == null) {
             return null;
         } else {
-            return resourceModel.getTags().stream().collect(Collectors.toMap(tag -> tag.getKey(), tag -> tag.getValue()));
+            return resourceModel.getTags().stream().collect(Collectors.toMap(Tag::getKey, Tag::getValue, (value1, value2) -> value2));
```
Comment (bradgr) [inline aws-ecr-repository/src/main/java/software/amazon/ecr/repository/Configuration.java:26]: Does this affect customer behavior at all? Can we test this through UpdateHandler tests?

File: aws-ecr-repository/src/test/java/software/amazon/ecr/repository/ConfigurationTest.java (hunk 1 above)
Comment (apurvmahajan) [reply aws-ecr-repository/src/test/java/software/amazon/ecr/repository/ConfigurationTest.java]: Yes. Updated as suggested.

File: aws-ecr-repository/src/main/java/software/amazon/ecr/repository/Configuration.java (hunk 2 above)
Comment (apurvmahajan) [reply aws-ecr-repository/src/main/java/software/amazon/ecr/repository/Configuration.java:26]: I had to add a unit test for this class. This part is abstracted in the UpdateHandler tests.

**Review by bradgr — APPROVED** [review]

**Review by lahirurd — APPROVED** [review]

**Pull request merged**